app/server/models/user.py

```diff
@@ -260,20 +260,18 @@ def location(self):
 
     @location.setter
     def location(self, location):
+        from server.utils.location import async_set_user_gps_from_location
 
         self._location = location
 
         if location is not None and location is not '':
-
-            if self.id is None:
-                raise AttributeError('User ID not set')
-
-            try:
-                task = {'user_id': self.id, 'address': location}
-                task_runner('worker.celery_tasks.geolocate_address', args=(task,))
-            except Exception as e:
-                sentry_sdk.capture_exception(e)
-                pass
+            # Delay execution until after request to avoid race condition with db
```

Comment:
What is the nature of the race condition?

Reply:
the thread needs to load in the user object from the db based off the user id. If that object hasn't been committed yet it won't exist